README.md

```diff
@@ -62,6 +60,76 @@ var app = builder.Build();
 ...
 ```
 
+## Table Considerations
+The cache provider will use the table specified by `TableName` if it already exists. The table must use a non-composite partition key of type string, otherwise an exception will be thrown during the first cache operation. No settings will be modified on existing tables.
+
+For production environments, we recommend setting `CreateTableIfNotExists` to `false` and providing both a `PartitionKeyName` and `TTLAttributeName`. This minimizes the required permissions and additional control plane calls during the first cache operation. It is recommended to only set `CreateTableIfNotExists` to `true` in development environments.
+
+When `CreateTableIfNotExists` is set to `true` and if the table specified by `TableName` does not already exist, the cache provider will create the table during the first cache operation with the following default settings:
+* The [on-demand](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.ReadWriteCapacityMode.html#HowItWorks.OnDemand)  capacity mode
+* [Time to Live](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/TTL.html) enabled
+* Encryption using an [AWS KMS key owned and managed by DynamoDB](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/encryption.howitworks.html#ddb-owned). If you would like to use an AWS managed or customer managed key, you can update the table after it is created or provide your own table. See [Managing encrypted tables](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/encryption.tutorial.html) in the Amazon DynamoDB Developer Guide.
+
+# Permissions
+To use the cache provider with an _existing_ DynamoDB table, the following permissions are required:
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "dynamodb:PutItem",
```

Comment:
Following the comment you added about the recommended is `CreateTableIfNotExists` to `false` and providing both a `PartitionKeyName` and `TTLAttributeName`. If they do that then the library does not require `dynamodb:DescribeTable` and `dynamodb:DescribeTimeToLive`. Can we remove those 2 permissions from this policy.

Reply:
You can add a sentence at the bottom of the snippet saying - " If the `PartitionKeyName` and `TTLAttributeName` are set, then the `dynamodb:DescribeTable` and `dynamodb:DescribeTimeToLive` permissions can be omitted.

Reply:
Addressed via https://github.com/awslabs/aws-dotnet-distributed-cache-provider/compare/2a4b15819408cd014c9bff1025fc0b3bccf13c0c..94d6ed5e7e59d3bbb28cfa1644dff55dfd15f4e0.

I added a third block for the middle case (existing table, but unknown key names)